gamemgr/bggame.cc

```diff
@@ -1889,23 +1890,82 @@ void BG_Game::end_game(bool success) {
 
 		// Fade out for 1 sec (50 cycles)
 		pal->fade(50, 0, 0);
-#if 0
-		//TODO: only when finishing a game and not when viewed from menu
-		if (when not in menu) {
-			if (wait_delay(10)) break;
 
-			// Congratulations
+		Game_clock *clock = gwin->get_clock();
+		int total_time = clock->get_total_hours();
+		// Congratulations screen
+		
+		// show only when finishing a game and not when viewed from menu
+		// game starts off with 6 hours as its 6am, should we be removing those 6hours?
+		// perhaps this should be gotten from the save game file creation date
+		if (total_time > 6) {
+			if (wait_delay(100)) {
+				throw UserSkipException();
+			}
 
 			// Paint backgound black
 			win->fill8(0);
 
 			starty = (gwin->get_height() - normal->get_text_height() * 6) / 2;
 
-			//TODO: figure out the time it took to complete the game
+			// calculate the time it took to complete the game
 			// in exultmsg.txt it is "%d year s ,  %d month s , &  %d day s"
 			// only showing years or months if there were any
 			for (unsigned int i = 0; i < 9; i++) {
 				message = get_text_msg(congrats + i);
+				// if we are on the line with the time played
+				if (i == 2) {
+					//initialize to make sure there is no garbage.
+					char buffer[50] = {0};
+					message = buffer;
+					// since we start at 6am we use hours if below 30 of them(30-6=1day), otherwise we don't display hours unless over a month so offset doesnt matter.
+					if (total_time >= 30) {
+						// this is how you would calculate years but since UltimaVII congrats screen has 13 months per year
+						// and VI in game calendar states 12 months per year
+						// it was decided to keep only months as we don't know which year is correct
+						// 8064 hours = 336 days, 672 hours = 28 days
+						// int year = total_time/8064;
+						// total_time %= 8064;
+						int month = total_time/672;
+						total_time %= 672;
+						int day = total_time/24;
+						total_time %= 24;
+						int hour = total_time;
+						// as per comment above
+						// if(year > 0) sprintf(buffer,"%d year(s) , ",year);
+						// message = buffer;
+						if(month == 1) sprintf(buffer,"%s%d month",message,month);
+						else if(month > 1) sprintf(buffer,"%s%d month's",message,month);
+						message = buffer;
+
+						// add amperstand only if month(s) and there is more to display.
+						if(day != 0 || hour != 0 && month > 0) sprintf(buffer,"%s & ",message);
```

Comment:
Please add a space between `if` and `(`., and break the `if` statement into its own line.